codegen/projections/rpcv2_cbor/lib/rpcv2_cbor/parsers.rb

```diff
@@ -29,14 +29,15 @@ def self.parse(list)
 
     # Error Parser for ComplexError
     class ComplexError
-      def self.parse(http_resp)
+      def self.parse(http_resp, **kwargs)
         data = Types::ComplexError.new
         body = http_resp.body.read
-        return data if body.empty?
-        map = Hearth::CBOR.decode(body)
-        data.top_level = map['TopLevel']
-        data.nested = (ComplexNestedErrorData.parse(map['Nested']) unless map['Nested'].nil?)
-        data
+        unless body.empty?
```

Comment:
Even if body is empty, wouldn't we parse an empty map easily here?

Reply:
I think in at least some cases we used to fail on this (xml?) - it also means you can skip any parsing + checking for each field in the map when you already know you don't need to do any more parsing.  We used to just return the data here immediately, so I think this new logic is equivalent to that + the error class wrapper.